compiler/rustc_codegen_llvm/src/back/archive.rs

```diff
@@ -24,6 +24,8 @@ use rustc_session::Session;
 pub struct LlvmArchiveBuilder<'a> {
     sess: &'a Session,
     additions: Vec<Addition>,
+
+    buffers: Vec<(String, Vec<u8>)>,
```

Comment:
If this PR goes towards merging, then this is better added as a new variant to `enum Addition` instead of a separate vector.